src/app/model/Rock.ts

```diff
@@ -0,0 +1,109 @@
+import { Board } from './Board';
+import { PieceMovement } from './PieceMovement';
+import { Side, Square } from './Types';
+
+export class Rock implements PieceMovement {
+  id: string;
+  side: Side;
+
+  constructor(id: string, side: Side) {
+    this.id = id;
+    this.side = side;
+  }
+
+  possibleMoves(position: Square, board: Board): Square[] {
+    let movesToGo: Square[] = [];
+    movesToGo.concat(this._goUp(position, board));
+    movesToGo.concat(this._goDown(position, board));
+    movesToGo.concat(this._goLeft(position, board));
+    movesToGo.concat(this._goRight(position, board));
+
+    return movesToGo;
+  }
+
+  _goUp(position: Square, board: Board): Square[] {
+    let movesToGo: Square[] = [];
+    let matrixDimension = board.squares[0].length;
```

Comment:
Jakie jest totalne MINIMUM, które potrzebujesz wiedzieć o Board w Twojej implementacji? Musisz wiedzieć, że to jest macierz :) ? Spróbuj opisać minimum koniecznej wiedzy, żeby ta implementacja działała. Na co Ci powinno odpowiadać Board?

Reply:
no dobra tu fakt, prosi się, żeby zrobić readOnly :-)

no z tym minimum to jest chyba problem ... no muszę na pewno wiedzieć na jakich polach co jest, więc informacja o tym, że to macierz raczej niezbędna - nie widzę innej możliwości iterowania po niej.....CHOCIAŻ
chyba, że board będzie miał metodę i zwracał mi informację co stoi na danym polu, a z poziomu Rock będę pytał o poszczególne pola... to miałeś na myśli?

Reply:
ale jakbym miał taką metodę checkSquare() w interfejsie Board to i tak ją bym musiał zaimplementować, a do niej będę się odwoływał co chwila, więc trochę to bez sensu -.- ...meh.

Reply:
Zdaje mi się że tak :) Skoro ważne jest dla Ciebie co stoi na danym polu to wtedy pozostawiasz wolność w tym jak ktoś to zaimplementuje pod spodem

Reply:
Nie musisz implementować - to jest właśnie najpiękniejsze :) Ktoś inny to zrobi a Ty zdefiniowałeś interfejs, że tak to ma działać. Teraz w teście możesz sobie zamockowac np. zwracanie danych pionków dla przekazanych pól. A pusta szachownica to tylko mock zwracający zawsze, że na danym polu nie ma pionka.